src/http/common.cr

```diff
@@ -82,7 +82,22 @@ module HTTP
     end
 
     io << "\r\n"
-    io << body if body
+
+    if body
+      if body.is_a?(IO)
+        if headers["Transfer-Encoding"] == "chunked"
```

Comment:
Done @jhass . With some other changes that were required.
Regarding the `Content-Length` I see that if the body is a IO, then chunked should be used, and hence, `Content-Length` must be avoided [ref](http://greenbytes.de/tech/webdav/rfc2616.html#rfc.section.4.4).